packages/text-annotator/src/highlight/highlightLayer.ts

```diff
@@ -7,11 +7,11 @@ import { trackViewport } from './trackViewport';
 import './highlightLayer.css';
 
 const debounce = <T extends (...args: any[]) => void>(func: T, delay: number = 10): T => {
-  let timeoutId: number;
+  let debounceTimer: ReturnType<typeof setTimeout> = undefined;
 
   return ((...args: any[]) => {
-    clearTimeout(timeoutId);
-    timeoutId = setTimeout(() => func.apply(this, args), delay);
+    clearTimeout(debounceTimer);
+    debounceTimer = setTimeout(() => func.apply(this, args), delay);
   }) as T;
```

Comment:
Fixed the typescript error with `Timeout` vs `number` confusion:
<img width="728" alt="image" src="https://github.com/recogito/text-annotator-js/assets/68850090/1f61eb39-61bc-4158-a5c3-68f2a429f693">

I saw that such a trick has already been used in the `SelectionHandler`:
https://github.com/recogito/text-annotator-js/blob/8ccace6254f599d25ebb6ddbac8c69fa571c1cf0/packages/text-annotator/src/SelectionHandler.ts#L66-L73